api/index.js

```diff
@@ -0,0 +1,92 @@
+require('dotenv').config();
+
+const config = require('./config');
+
+const { sumVotes } = require('./utils');
+
+const { db, writeTalk, addVote, auth } = require('./db');
+
+const express = require('express');
+const bodyParser = require('body-parser')
+const app = express();
+
+const authHeaderToUserId = async authHeader => {
+  const m = authHeader && authHeader.match(/Bearer (.+)/);
+
+  if (!m) throw new Error('Request missing ID token');
+
+  const idToken = m[1];
+  const decodedToken = await auth(idToken);
+
+  return decodedToken.user_id;
+};
+
+app.use((req, res, next) => {
+  res.header('Access-Control-Allow-Origin', '*');
+  res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept, Authorization');
+  next();
+});
+
+app.use(bodyParser.json());
+
+app.get('/talks', async (req, res) => {
+  const limit = +req.query.limit || config.DEFAULT_PAGE_LENGTH;
+  const after = +req.query.after || 0;
```

Comment:
I was previously in the process of adding pagination, but decided against it because the data dynamically changes sequence based on user actions, making pagination a  less-than-ideal solution. I should have caught the unused variables, though.

If this was an app with a lot of traffic and hundreds/thousands of talks to choose from with frequent real-time updates to votes, I'd probably use websockets to get the real-time data and update the sequence based on that. I'd also implement infinite scrolling (makes more sense than pagination for this use case) with markers indicating the ranking in the list.